Feature/jslogin -> Created JS functionality for LOGIN validation, an password encryption method in util, and SKELETON handler for login and reg

sumanyuj opened the pull request:
Summary of changes 

## index.html
Created a function using inline script that in essence calls a django view method that calls Alex's password validator method. I have some text show up (a div elem) that states if a password meets the reqs in green (or not, in red). 

## views.py and urls.py
Created a method to handle POST request for aforementioned validation method, using JSON. Created two skeleton methods to handle redirects and (in the future, when Eric merges his commits) implement functionality WITHIN those methods to encrypt those passes and store them (and usernames) in POSTGRES. Of course, everything must have a matching pattern in urls.py, which is what I did.

## eg_util/encrypt.py
Added a simple method to take a plaintext password and salt and hash it through bcrypt. 

## In the future:
1) This is before Rances merge, so we may need to refactor the paths
2) @csrf_exempt. We need to implement CSRF tokens. This is a placeholder to get it to work for now, for not just the validator method but for the rest of the functions. 

Closes #19

sumanyuj committed:
added onclick property to html form for passwords to connect to password validator, created function (js) inline to changer color of url bar depending on if the password validator meets

sumanyuj committed:
added pattern for password validator in urls.py and function to plumb together Alex's method in views.py

sumanyuj committed:
working commit for checking if the password is validated

WeirdAlex03 commented on evergreen/eg_app/tests/util/test_validators.py:
```diff
@@ -1,5 +1,5 @@
 from django.test import TestCase
-from eg_app.util.validators import validate_email, validate_password
+from eg_app.eg_util.validators import validate_email, validate_password
```
The `eg_app/tests/*` subdirectories should mirror `eg_app/*`. If `eg_app/util` is renamed, `eg_app/tests/eg_util` should also be updated

WeirdAlex03 commented on evergreen/eg_app/views.py:
```diff
@@ -1,10 +1,25 @@
+@csrf_exempt
+def validate_pass(request):
+    if request.method == "POST":
+        password = request.POST.get("password")
+        # this uses the method from Alex's Method
+        isValid = val.validate_password(password)
+        print(isValid)
+        return JsonResponse({"valid_pass":str(isValid)})
```
Email should be validated too. Renaming this route to just `validate`, that can be added to the same JSON Response

WeirdAlex03 commented on evergreen/index.html:
```diff
@@ -49,6 +51,47 @@ <h2>Contact Us</h2>
+                else
+                {
+                    messageDiv.style.color = "red"
+                    messageDiv.textContent = "BAD PASSWORD"
```
This should all be in the Javascript file and we should explicitly say what the rule is (tell user they need >=12 char)

sumanyuj replied:
Done

WeirdAlex03 reviewed (changes requested):
Overall good, a few small comments that should be quick to implement. Ran it locally and it works fine

Could you clarify why you renamed `util`->`eg_util`? `eg_app` was just the name of the service, I don't think we hvae to pull a McDonalds and McName everything with an `eg_` (unless there's another reason I don't see)

sumanyuj commented:
I renamed it as an artifact of trying to get it to actually work by calling your validators method. I thought maybe that it conflicted with another package named utils or whatever, but that turned out not be the case.

I'll rename it back

WeirdAlex03 commented:
One other thing, I just realized you mentioned encrypt.py in your comment, but didn't commit it. Git doesnt track new files by default; if youre on the command line you'll have to `git add` it

sumanyuj commented:
> One other thing, I just realized you mentioned encrypt.py in your comment, but didn't commit it. Git doesnt track new files by default; if you're on the command line you'll have to `git add` it

weird, I thought I did. Good catch

sumanyuj committed:
added encrypt function

sumanyuj committed:
changed a few wordings

sumanyuj committed:
fixed issues with seeing validation text

sumanyuj commented:
Ok, so that took longer than I thought, in part because I was trying to make the in line script to function.js. Welp, turns out I can't use my beloved Django syntax (await fetch("{% url 'validate' %}") in function.js, a standalone JS file. 

Provided enough time, I could refactor it (and probably will in future) but for the sake of finishing I am keeping it inline for now.

I addressed some more pressing issues, such as adding email validation, CSRF tokens (not really tested yet, but most of the code is there), and renamed eg_util. Also fixed encrypt issue.

-sj

WeirdAlex03 reviewed (approved):
There are some formatting problems that I must complain about. And now that I have satisfied that requirement, lgtm shipit :shipit:

(we'll deal with those later in the Part 2 sprint when Eric adds a linter/formatter GitHub Action)

giagermain reviewed (approved):
LGTM
